src/base/controller.hpp

```diff
@@ -217,6 +233,15 @@ void Controller::set_config(const json_t &config) {
   if (max_parallel_experiments_ > 1)
     max_parallel_shots_ = 1;
  
+  if (JSON::check_key("max_statevector_memory_mb", config)) {
+    JSON::get_value(max_statevector_memory_mb_, "max_statevector_memory_mb", config);
+  } else {
+    int system_memory_mb = (int) get_system_memory_mb();
+    max_statevector_memory_mb_ = 1;
+    while((max_statevector_memory_mb_ << 1) < system_memory_mb)
+      max_statevector_memory_mb_ = max_statevector_memory_mb_ << 1;
```

Comment:
`system_memory_mb` is not free memory. It is system memory. If we set `max_statevector_memory_mb_` to the `system_memory_mb`, thrashing will occurs frequently.

I just use the max of power of two as the default value in the system memory. But this may not be clear specification.

How about half of the system memory as the default `max_statevector_memory_mb_` ?

Reply:
So I think that we have to choose between thrashing vs not executing at all.
I'd vote for thrashing, even though it can be very painful on larger experiments... what do you think?

Reply:
I believe that we should avoid thrashing. To accelerate noise simulation (even though its qubit is small), qasm and statevector simulator will use memory as much as possible to increase a number of threads. If application does not specify `max_statevector_memory_mb_`, users will encounter thrashing frequently.

In case of jdk8u, JVM uses half of the system memory if the system memory is less or equal to 248MB, otherwise quarter.

Reply:
Ok, after talking with Chris, he agrees with @hhorii so let's not start the simulation if thrashing can kick in.